src/plugins/plugin-http.ts

```diff
@@ -42,6 +42,16 @@ function getSpanName(options: ClientRequestArgs|url.URL) {
   return options.hostname || options.host || 'localhost';
 }
 
+/**
+ * Returns whether the Expect header is on the given options object.
+ * @param options Options for http.request.
+ */
+function hasExpectHeader(options: ClientRequestArgs|
+                         url.URL): options is ClientRequestArgs {
```

Comment:
Should the return type a `boolean`?

Reply:
`A is B` is already a `boolean`.

Reply:
Yeah, I know. What I mean here is that this function does NOT test that `options` is a `ClientRequestArgs` object but rather that it is a one AND it has an `Expect` header. Right?

Reply:
Forgot to mention that this makes the compiler make bad decisions about type info. E.g.
```ts
const options = ...;
...
if (hasExpectHeader(options)) {
  // Compiler thinks `options` is a ClientRequestArgs, which is correct.
} else {
  // Compiler thinks `options` is NOT a ClientRequestArgs, which may or may not be correct.
}
```